ddtrace/tracer/tracer.go

```diff
@@ -268,6 +291,42 @@ func newTracer(opts ...StartOption) *tracer {
 		t.reportHealthMetrics(statsInterval)
 	}()
 	t.stats.Start()
+	// Start collecting telemetry data
+	go func() {
```

Comment:
I didn't document it well, but the client's `Start` method is non-blocking. It does a little bit of work up-front, but then kicks off background workers and returns. So it won't delay starting a tracer too much. It also copies its input arguments, which should hopefully get rid of the race detector failures.

Reply:
Ah ok, thanks for the info!